iceoryx_hoofs/test/moduletests/test_vocabulary_semantic_string.cpp

```diff
@@ -0,0 +1,367 @@
+{
+    ::testing::Test::RecordProperty("TEST_ID", "14483f4e-d556-4770-89df-84d873428eee");
+    using SutType = typename TestFixture::SutType;
+
+    for (auto& value : TestValues<SutType>::INVALID_CHARACTER_VALUES)
+    {
+        auto sut = SutType::create(string<SutType::capacity()>(TruncateToCapacity, value.c_str()));
+
+        ASSERT_THAT(sut.has_error(), Eq(true));
+        ASSERT_THAT(sut.get_error(), Eq(SemanticStringError::ContainsInvalidCharacters));
+    }
+}
+
+TYPED_TEST(SemanticString_test, InitializeWithStringContainingIllegalContentFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "9380f932-527f-4116-bd4f-dc8078b63330");
+    using SutType = typename TestFixture::SutType;
+
+    for (auto& value : TestValues<SutType>::INVALID_CONTENT_VALUES)
+    {
+        auto sut = SutType::create(string<SutType::capacity()>(TruncateToCapacity, value.c_str()));
+
+        ASSERT_THAT(sut.has_error(), Eq(true));
+        ASSERT_THAT(sut.get_error(), Eq(SemanticStringError::ContainsInvalidContent));
+    }
+}
+
+TYPED_TEST(SemanticString_test, InitializeWithTooLongContentFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "b5597825-c559-48e7-96f3-5136fffc55d7");
+    using SutType = typename TestFixture::SutType;
+
+    for (auto& value : TestValues<SutType>::TOO_LONG_CONTENT_VALUES)
+    {
+        auto sut = SutType::create(string<SutType::capacity() * 2>(TruncateToCapacity, value.c_str()));
+
+        ASSERT_THAT(sut.has_error(), Eq(true));
+        ASSERT_THAT(sut.get_error(), Eq(SemanticStringError::ExceedsMaximumLength));
+    }
+}
+
+TYPED_TEST(SemanticString_test, AppendValidContentToValidStringWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "0994fccc-5baa-4408-b17e-e2955439608d");
+    using SutType = typename TestFixture::SutType;
+
+    for (auto& value : TestValues<SutType>::VALID_VALUES)
+    {
+        for (auto& add_value : TestValues<SutType>::VALID_VALUES)
+        {
+            auto sut = SutType::create(string<SutType::capacity()>(TruncateToCapacity, value.c_str()));
+            ASSERT_THAT(sut.has_error(), Eq(false));
+
+            EXPECT_THAT(sut->append(string<SutType::capacity()>(TruncateToCapacity, add_value.c_str())).has_error(),
+                        Eq(false));
+            auto result_size = value.size() + add_value.size();
+            EXPECT_THAT(sut->size(), result_size);
+            EXPECT_THAT(sut->capacity(), Eq(TestValues<SutType>::CAPACITY));
+
+            auto result = value + add_value;
+            EXPECT_THAT(sut->as_string().c_str(), StrEq(result));
+        }
+    }
+}
+
+TYPED_TEST(SemanticString_test, AppendInvalidContentToValidStringFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "fddf4a56-c368-4ff0-8727-e732d6ebc87f");
+    using SutType = typename TestFixture::SutType;
+
+    for (auto& value : TestValues<SutType>::VALID_VALUES)
+    {
+        for (auto& invalid_value : TestValues<SutType>::INVALID_CHARACTER_VALUES)
```

Comment:
The test name does not match with the test values. `InvalidContent` vs `INVALID_CHARACTER`. I think you need to duplicate this test case to have both cases tested.

Reply:
Here and also in the insert test case I adjusted the test name to `AppendInvalidCharactersToValidStringFails`

The reason is that adding invalid characters will fail but adding invalid content most likely will succeed. Here an example:
1. A file name is not allowed to contain `.` or `..` but adding `.` to `hello` results in the valid file name `hello.`
2. Users are not allowed to start with dash, so `-shumway` is illegal content but adding or inserting this content results in the legal name `gordon-shumway`.

Reply:
Ah, okay. It is just that the path is not tested according to codecov